File: src/sum_single.rs
// sum_single.rs - Rust single-threaded sum
use std::time::Instant;

fn main() {
    let start = Instant::now();
    
    let n: i64 = 100_000_000;
    let mut sum: i64 = 0;
    
    for i in 1..=n {
        sum += i;
    }
    
    let elapsed = start.elapsed();
    println!("Sum: {}", sum);
    println!("Rust Single-threaded Time: {:?}", elapsed);
    
    // Verify correctness
    let expected: i64 = 5000000050000000;
    if sum == expected {
        println!("✅ Result is correct!");
    } else {
        println!("❌ Expected {}, got {}", expected, sum);
        std::process::exit(1);
    }
}

File: src/sum_single.go
// sum_single.go - Go single-threaded sum
package main

import (
    "fmt"
    "time"
)

// RunSingleSum executes the single-threaded sum logic.
func RunSingleSum() {
    start := time.Now()
    
    var sum int64 = 0
    const n = 100_000_000
    
    for i := int64(1); i <= n; i++ {
        sum += i
    }
    
    elapsed := time.Since(start)
    fmt.Printf("Sum: %d\n", sum)
    fmt.Printf("Go Single-threaded Time: %v\n", elapsed)
    
    // Verify correctness
    const expected int64 = 5000000050000000
    if sum == expected {
        fmt.Println("✅ Result is correct!")
    } else {
        fmt.Printf("❌ Expected %d, got %d\n", expected, sum)
    }
}